inventory/src/main/java/com/yas/inventory/service/StockService.java

```diff
@@ -50,4 +71,50 @@ public void addProductIntoWarehouse(List<StockPostVM> postVMs) {
 
     }
 
+    public List<StockVm> getStocksByWarehouseIdAndProductNameAndSku(Long warehouseId,
+                                                                    String productName,
+                                                                    String productSku) {
+        HashMap<Long, ProductInfoVm> productInfoVmHashMap = (HashMap<Long, ProductInfoVm>) warehouseService.getProductWarehouse(
+                        warehouseId,
+                        productName,
+                        productSku,
+                        FilterExistInWHSelection.YES)
+                .stream()
+                .collect(Collectors.toMap(ProductInfoVm::id, productInfoVm -> productInfoVm));
+
+        List<Stock> stocks = stockRepository.findByWarehouseIdAndProductIdIn(
+                warehouseId,
+                productInfoVmHashMap.values().stream().map(ProductInfoVm::id).toList()
+        );
+
+        return stocks.stream().map(
+                stock -> {
+                    ProductInfoVm productInfoVm = productInfoVmHashMap.get(stock.getProductId());
+
+                    return StockVm.fromModel(stock, productInfoVm);
+                }
+        ).toList();
+    }
+
+    public void updateProductQuantityInStock(final StockQuantityUpdateVm requestBody) {
+        List<StockQuantityVm> stockQuantityVms = requestBody.stockQuantityList();
+        List<Stock> stocks = stockRepository.findAllById(stockQuantityVms.stream().map(StockQuantityVm::stockId).toList());
+
+        for (final Stock stock : stocks) {
```

Comment:
I think this flow does not need to be executed in order so we can use parallel stream to increase performance?